sky/provision/kubernetes/instance.py

```diff
@@ -253,6 +253,31 @@ def _run_command_on_pods(node_name, node_namespace, command):
         _request_timeout=kubernetes.API_TIMEOUT)
     return cmd_output
 
+def _run_command_on_pods_with_logs(node_name, node_namespace, command, command_list):
+    cmd_output = kubernetes.stream()(
+        kubernetes.core_api().connect_get_namespaced_pod_exec,
+        node_name,
+        node_namespace,
+        command=command,
+        stderr=True,
+        stdin=True,
+        stdout=True,
+        tty=False,
+        _preload_content=False,
+        _request_timeout=kubernetes.API_TIMEOUT)
+    while cmd_output.is_open():
+        cmd_output.update(timeout=1)
```

Comment:
What's the tradeoff of setting a lower/higher timeout here?

Reply:
`_request_timeout=kubernetes.API_TIMEOUT` is a standard parameter we have defined (https://github.com/shethhriday29/skypilot/blob/a0509c927b4f15c8fb129286b6626beed3fc4e8d/sky/adaptors/kubernetes.py#L22-L23), so just sticking to the default here unless you want otherwise!

Reply:
Ah sorry, I mean the `timeout` in `cmd_output.update(timeout=1)`. What happens if we set it lower/higher?